### PR TITLE
Upgrade hostdata

### PR DESCRIPTION
**Pull request by jmhooper**
**Why**: So that we get all of the latest and greatest goodies from Identity::Hostdata

**How**: The only consumer of Hostdata was the activate script. We are moving away from that so I went ahead and moved the IdentityConfigReader tooling from the IDP into this repo.

### CODE REVIEW

File: lib/identity_config_reader.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @logger = logger
+    @s3_client = s3_client
+  end
+
+  def read_configuration(write_copy_to: nil)
+    configuration = read_default_configuration.deep_merge(
+      read_override_configuration,
+    )
+
+    if write_copy_to && !File.exists?(write_copy_to)
```
Comment (zachmargolis) [inline lib/identity_config_reader.rb:19]: 😬 can we pull this into the gem? or is it urgent to get this merged before that?

Comment (zachmargolis) [reply lib/identity_config_reader.rb:19]: not stopping until we have `identity-hostdata@999999.99999`

Comment (jmhooper) [reply lib/identity_config_reader.rb:19]: Hostdata PR: https://github.com/18F/identity-hostdata/pull/26

**Comment by jmhooper** [comment]
PR is updated to use the hostdata ConfigReader. Right now it is using a branch version. I'll switch to the new tag before merging.

**Review by zachmargolis — APPROVED** [review]
LGTM

**Pull request merged**